deepchecks/recommender/checks/product_association.py

```diff
@@ -0,0 +1,165 @@
+            filtered_interactions[f'delta_{datetime_name}_seconds'] = (filtered_interactions[datetime_name] - filtered_interactions[f'prev_{datetime_name}']).dt.total_seconds()
+            filtered_interactions = filtered_interactions[filtered_interactions['delta_timestamp_seconds'] < self.max_timestamp_delta].drop([f'prev_{datetime_name}',f'{datetime_name}'],axis=1)
+                 
+        products_association_df = filtered_interactions[[f'prev_{item_col}', item_col]].copy()                                                                                                                               
+        products_association_df['item_combination'] = products_association_df.apply(lambda x: tuple(sorted(x)), axis=1)
+
+        # Create a new DataFrame to count the non-directional co-occurrence of each pair
+        association_counts = products_association_df.groupby(['item_combination']).size().reset_index(name='co-occurence')
+        # Use the apply method to unpack the tuples into separate columns
+        association_counts[f'prev_{item_col}'], association_counts[item_col] = zip(*association_counts['item_combination'])
+        # Drop the original 'tuple_column' 
+        association_counts.drop(columns=['item_combination'], inplace=True)
+
+        prob_df = filtered_interactions.groupby(item_col).size().reset_index(name='popularity')
+        prob_df['probability'] = prob_df['popularity']/len(filtered_interactions)
+        prob_dict = prob_df.set_index(item_col)['probability'].to_dict()
+        
+        # Function to calculate probability of an item
+        def get_probability(item):
+            return prob_dict.get(item, 0)
+
+        # Add popularity columns to the DataFrame
+        association_counts[f'prev_{item_col}_prb'] = association_counts[f'prev_{item_col}'].apply(get_probability)
+        association_counts[f'{item_col}_prb'] = association_counts[item_col].apply(get_probability)
+        association_counts['lift_metric (%)'] = association_counts['co-occurence']/(association_counts[f'prev_{item_col}_prb']*association_counts[f'{item_col}_prb']*len(filtered_interactions)).astype(np.float16)
+        
+        association_counts = association_counts[[f'prev_{item_col}', item_col,'co-occurence','lift_metric (%)']]
+        movie_translation = item_df[[item_col,item_column_name]].set_index(item_col)[item_column_name].to_dict()
+        
+        association_counts[f'prev_{item_col}']= association_counts[f'prev_{item_col}'].map(movie_translation)
+        association_counts[item_col]= association_counts[item_col].map(movie_translation)
+
+        association_counts = association_counts.sort_values(['co-occurence','lift_metric (%)'],ascending=[False,False]).rename({f'prev_{item_col}':'reference',
+        item_col:'possible recommandation'},axis=1)
+        display_df = association_counts.head(20)
+        # Create a network graph
+        G = nx.Graph()
```

Comment:
Add to requirements

Reply:
ba916c4d506e23113ed7a7464e953ca98c2c649e
Need to fix the conflit